### ISSUE TITLE
NumberInput when focused does not change its value on rerender

### ISSUE THREAD

**Issue by skelawsky** [mantinedev/mantine#3059]
### What package has an issue

@mantine/core

### Describe the bug

I have an Input component created, onBlur patch is sent to the backend, new data is received. This is where the rerender of all inputs takes place. The second input value depends on the first one. If I manually (i.e., click outside the first input being changed, e.g., on div) the values in the second input change. On the other hand, if I move the focus to the second input (by click or tab) then the value remains the same - which is an error. How to fix this?

```
<NumberInput
  key={info.cell.id}
  value={parseInt(info.getValue())}
  onBlur={(e) => {
    updateProduct({
      data: { [info.cell.column.id]: e.target.value },
      id: info.cell.row.original.id,
    });
  }}
/>
```

[YouTube video showing bug](https://youtu.be/qXUR_s2oTYw)

### What version of @mantine/hooks page do you have in package.json?

5.7.2

### Do you know how to fix the issue

No

### Are you willing to participate in fixing this issue and create a pull request with the fix

Yes

**Comment by rtivital** [comment]
It is how NumberInput component works – value is not updated when the input is focused. Otherwise, there will be infinite rerenders. I do not know any good ways of fixing that.

**Comment by tombohub** [comment]
Chakra has the same problem except slightly different. In my case Chakras last input is not updated: https://github.com/chakra-ui/chakra-ui/issues/7086 and Mantines first input is not updated: https://github.com/mantinedev/mantine/issues/3145

**Comment by rtivital** [comment]
Fixed in 6.0
